docs/pages/preview/custom-build-config-schema.mdx

```diff
@@ -0,0 +1,438 @@
+---
+title: EAS Build custom builds config schema
+description: A reference of configuration options for custom workflows with EAS Build.
+hideFromSearch: true
+---
+
+Creating custom workflows for EAS Build helps customize the build process by adding your workflows.
+
+## YAML syntax for workflows
+
+Workflow files are stored inside the **.eas/build** directory path. They use YAML syntax and must have a `.yml` or `.yaml` file extension. If you are new to YAML or want to learn more about the syntax, see [Learn YAML in Y minutes](https://learnxinyminutes.com/docs/yaml/).
+
+## `build`
+
+Defined to describe a custom workflow. All config options to create a workflow are specified under it.
+
+### `name`
+
+The name of your workflow that is used to identify the workflow in the build logs. EAS Build uses this property to display the name of your workflow in the dashboard.
+
+For example, the workflow's name is "Run tests":
+
+```yaml
+build:
+  # @info #
+  name: Run tests
+  # @end #
+  steps:
+    - run:
+        name: Install dependencies
+        command: npm install
+```
+
+### `steps`
+
+Steps are used to describe a list of actions either in the form of commands or functions. These actions are executed when a workflow is triggered. You can define single or multiple steps in a workflow. It is **required** to define at least one step per workflow.
+
+Each step is configured with the following properties:
+
+#### `steps[].run`
+
+The `run` key is used to trigger a set of instructions. For example, a `run` key is used to install dependencies using the `npm install` command:
+
+```yaml
+build:
+  name: Install npm dependencies
+  steps:
+    # @info #
+    - run:
+        name: Install dependencies
+        command: npm install
+    # @end #
+```
+
+You can also use `steps.run` to execute inline bash shell commands:
+
+```yaml
+build:
+  name: Run inline shell commands
+  steps:
+    # @info #
+    - run: echo "Hello world"
+    # @end #
+```
+
+#### Use a single step
+
+For example, a workflow with the following `steps` will run a single command to install npm dependencies:
+
+```yaml
+build:
+  name: Install npm dependencies
+  steps:
+    # @info #
+    - run:
+        name: Install dependencies
+        command: npm install
+    # @end #
+```
+
+> **Note:** `-` before `run` counts as indentation.
+
+#### Use multiple steps
+
+When multiple `steps` are defined, they are executed sequentially. For example, a workflow with the following `steps` will first run a single command to install npm dependencies and then run a command to run tests:
+
+```yaml
+build:
+  name: Run tests
+  steps:
+    # @info #
+    - run:
+        name: Install dependencies
+        command: npm install
+    - run:
+        name: Run tests
+        command: |
+          echo "Running tests..."
+          npm test
+    # @end #
+```
+
```

Comment:
Please add a section about `steps[].run.id`.

There are two use cases for that: 
1. Calling the same function that produces an output/outputs multiple times.
2. Using output from a step (not a function call!) in another step.

(1)
Let's say you have a function for generating a random value:
```
  random:
    name: Generate random number
    outputs: [value]
    command: set-output value `random_number`
```
You want to generate two random number and print them in a step:
```

build:
  name: Functions Demo
  steps:
    - random:
        id: random_1
    - random:
        id: random_2
    - run:
        name: Print random numbers
        inputs:
          random_1: ${ steps.random_1.value }
          random_2: ${ steps.random_2.value }
        command: | 
          echo "${ inputs.random_1 }"
          echo "${ inputs.random_2 }"
```


(2)
You have a step that produces an output and you want to access it in another step:

```
    - run:
        name: Produce output
        id: id123    # <---- !!!
        outputs: [foo]
        command: |
          echo "Producing output for another step"
          set-output foo bar
    - run:
        name: Use output from another step
        inputs:
          foo: ${ steps.id123.foo }
        command: |
          echo "foo = \"${ inputs.foo }\""
```